data/src/main/java/com/linkedin/data/schema/PathSpecSet.java

```diff
@@ -0,0 +1,440 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.data.schema;
+
+import com.linkedin.data.template.RecordTemplate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+
+/**
```

Comment:
For all javadocs in this class, I'd suggest providing examples using the serialized pathspec syntax (e.g. `/foo/bar`, `/foo`) for clarity, especially in more complicated examples (e.g. copy with scope, remove prefix)

Reply:
I edited the javadoc, let me know if this is what you had in mind

Reply:
Adding `/foo` directly into code segments may be confusing, as it's not proper Java syntax. However, I still think it's more clear to users. I originally meant adding this in addition to the code examples, like `set of {/a} contains /a` or something like that.